java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -371,4 +381,56 @@ public CompletableFuture<Long> exists(@NonNull String[] keys) {
     public CompletableFuture<Long> unlink(@NonNull String[] keys) {
         return commandManager.submitNewCommand(Unlink, keys, this::handleLongResponse);
     }
+
+    @Override
+    public CompletableFuture<Long> zadd(
+            @NonNull String key,
+            @NonNull Map<String, Double> membersScoresMap,
+            @NonNull ZaddOptions options,
+            boolean changed) {
+        String[] changedArg = changed ? new String[] {"CH"} : new String[] {};
+        String[] membersScores = convertMapToValueKeyStringArray(membersScoresMap);
+
+        String[] arguments =
+                concatenateArrays(new String[] {key}, options.toArgs(), changedArg, membersScores);
+
+        return commandManager.submitNewCommand(Zadd, arguments, this::handleLongResponse);
+    }
+
+    @Override
+    public CompletableFuture<Long> zadd(
+            @NonNull String key,
+            @NonNull Map<String, Double> membersScoresMap,
+            @NonNull ZaddOptions options) {
+        return this.zadd(key, membersScoresMap, options, false);
+    }
+
+    @Override
+    public CompletableFuture<Long> zadd(
+            @NonNull String key, @NonNull Map<String, Double> membersScoresMap, boolean changed) {
+        return this.zadd(key, membersScoresMap, ZaddOptions.builder().build(), changed);
+    }
+
+    @Override
+    public CompletableFuture<Long> zadd(
+            @NonNull String key, @NonNull Map<String, Double> membersScoresMap) {
+        return this.zadd(key, membersScoresMap, ZaddOptions.builder().build(), false);
+    }
+
+    @Override
+    public CompletableFuture<Double> zaddIncr(
+            @NonNull String key, @NonNull String member, double increment, @NonNull ZaddOptions options) {
+        String[] arguments =
+                concatenateArrays(
+                        new String[] {key},
+                        options.toArgs(),
+                        new String[] {"INCR", Double.toString(increment), member});
```

Comment:
Consider making "INCR" a constant that you can reuse in test.

Reply:
It is going to get mixed with RequestOuterClass Incr would rather not.